services/workflows-service/src/document/document.service.ts

```
@@ -588,34 +601,46 @@ export class DocumentService {
         });
       }),
       individuals: {
-        ubos: entitiesWithDocuments.ubos.flatMap(ubo =>
-          parsedUIDocuments.individuals.ubos.map(expectedDoc =>
-            generateDocumentTrackerItem(
-              ubo.documents.find(doc => isMatchingDocument(doc, expectedDoc)),
-              expectedDoc,
-              {
-                id: ubo.id,
-                firstName: ubo.firstName,
-                lastName: ubo.lastName,
-                entityType: 'ubo',
-              },
-            ),
-          ),
-        ),
-        directors: entitiesWithDocuments.directors.flatMap(director =>
-          parsedUIDocuments.individuals.directors.map(expectedDoc =>
-            generateDocumentTrackerItem(
-              director.documents.find(doc => isMatchingDocument(doc, expectedDoc)),
-              expectedDoc,
-              {
-                id: director.id,
-                firstName: director.firstName,
-                lastName: director.lastName,
-                entityType: 'director',
-              },
-            ),
-          ),
-        ),
+        ubos: parsedUIDocuments.individuals.ubos.map(parsedDocument => {
+          const { ballerineEntityId } = parsedDocument;
+          const ubo = entitiesWithDocuments.ubos.find(ubo => ubo.id === ballerineEntityId);
+
+          if (!ubo) {
+            throw new Error('Ubo not found');
+          }
```

Don't think we should fail all the other ubos if one ubo was not found.